src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,125 @@
+
+            if (string.IsNullOrWhiteSpace(compiledFilePath))
+            {
+                throw new ArgumentException($"Invalid input file");
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                AppendDiagnosticsMessage(diagnosticsByFile, bicepFile, sb);
+                return sb.ToString();
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite))
+            {
+                TemplateEmitter emitter = new TemplateEmitter(semanticModel, ThisAssembly.AssemblyFileVersion);
+                EmitResult result = emitter.Emit(fileStream);
+
+                sb.AppendLine("Build succeeded. Created transpiled ARM template: " + compiledFile);
+            }
+
+            return sb.ToString();
+        }
+
+        private void AppendDiagnosticsMessage(KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile, string bicepFile, StringBuilder sb)
+        {
+            sb.AppendLine("Build failed. Please fix below errors in " + bicepFile + " :");
+
+            IEnumerable<IDiagnostic> diagnostics = diagnosticsByFile.Value;
+            IReadOnlyList<int> lineStarts = diagnosticsByFile.Key.LineStarts;
+
+            for (int i = 0; i < diagnostics.Count(); i++)
+            {
+                IDiagnostic diagnostic = diagnostics.ElementAt(i);
+                var (startLine, startChar) = TextCoordinateConverter.GetPosition(lineStarts, diagnostic.Span.Position);
+
+                sb.AppendLine((i+1) + ". " + diagnostic.Message + " bicep(" + diagnostic.Code + "). " + "[" + (startLine + 1) + ", " + (startChar + 1) + "]");
+            }
+        }
+
+        private string? GetCompiledFilePath(string bicepFilePath, StringBuilder sb, out string bicepFile, out string compiledFile)
+        {
+            compiledFile = string.Empty;
+            bicepFile = Path.GetFileName(bicepFilePath);
+
+            string? bicepFileWithoutExtension = Path.GetFileNameWithoutExtension(bicepFilePath);
+            string? folder = Path.GetDirectoryName(bicepFilePath);
+
+            if (string.IsNullOrWhiteSpace(bicepFileWithoutExtension) || string.IsNullOrWhiteSpace(folder))
+            {
+                return null;
+            }
+
+            sb.AppendLine("Build started...");
```

Comment:
>started [](http://example.com/codeflow?start=33&length=7)

The contents of the string builder are returned all at once back to the LSP client, so the user will never see it show up with the diagnostics and the final messages. Is there any value to logging it?

Reply:
Removed the logging:
https://github.com/Azure/bicep/commit/daa24e89c632c399b92f0de41c98fcea77b930eb